### ISSUE TITLE
FIle upload in Internet Explorer stopped working in 3.5.1

### ISSUE THREAD

**Issue by jochenberger** [SeleniumHQ/selenium#4497]
Using remote tests with selenium-server-standalone 3.5.1 with IEDriverServer 3.5.1 on Windows 10, I get
```
java.lang.ClassCastException: com.google.common.collect.Maps$TransformedEntriesMap cannot be cast to java.lang.String
	at org.openqa.selenium.remote.RemoteWebElement.upload(RemoteWebElement.java:110)
	at org.openqa.selenium.remote.RemoteWebElement.sendKeys(RemoteWebElement.java:95)
	at geb.navigator.NonEmptyNavigator.setInputValue(NonEmptyNavigator.groovy:699)
...
```

**Comment by p0deje** [comment]
Please provide [concise reproducible test case](http://sscce.org/) so that we could act on this issue.

See [CONTRIBUTING.md](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#issue-contributions)

**Comment by jochenberger** [comment]
That's IE 11.540.15063.0 btw.

**Comment by jochenberger** [comment]
I suspect it's related to https://github.com/SeleniumHQ/selenium/commit/e60b607a97b9b7588d59e0c26ef9a6d1d1350911

**Comment by p0deje** [comment]
We have tests for file uploading in IE and they pass. Please, provide a script that we can run locally to demonstrate the bug.

**Comment by jochenberger** [comment]
Do those tests use a RemoteDriver / FileDetector?
The ClassCastException is apparently caused by the fact that the standalone server tries to respond with an error message but the RemoteWebDriver expects a String in https://github.com/SeleniumHQ/selenium/blob/selenium-3.5.1/java/client/src/org/openqa/selenium/remote/RemoteWebElement.java#L110.
The original error that occurs on the Windows machine is this:
```
To downstream: { "error" : "Unknown method", "message" : "Command not found: POST /session/9804e13b-efe2-49af-8fba-ed4ac0868198/file" }
```

**Comment by jochenberger** [comment]
Apparently it's not an IE-only issue. I'm also getting this with Firefox (only if marionette is enabled):
```
org.openqa.selenium.UnsupportedCommandException: POST /session/826e0378-c6a3-4fc5-bc4a-33f8482ea4ab/file did not match a known command
        at org.openqa.selenium.remote.http.W3CHttpResponseCodec.createException(W3CHttpResponseCodec.java:185)
        at org.openqa.selenium.remote.http.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:120)
        at org.openqa.selenium.remote.http.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:49)
        at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:164)
        at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:641)
        at org.openqa.selenium.remote.RemoteWebElement.execute(RemoteWebElement.java:275)
        at org.openqa.selenium.remote.RemoteWebElement.upload(RemoteWebElement.java:109)
        at org.openqa.selenium.remote.RemoteWebElement.sendKeys(RemoteWebElement.java:95)
        at geb.navigator.NonEmptyNavigator.setInputValue(NonEmptyNavigator.groovy:699)
```
But in this case, the `execute` call throws the exception (in `RemoteWebElement.java:109`), it doesn't even get to the cast in line 110.
So there are actually 2 issues. The failing upload, which affects both IE and Firefox, and the failing error reporting.

**Comment by barancev** [comment]
Fixed in version 3.5.2 of the server
